pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,373 @@
+	       if netinfo.InterfaceType == sriovInterfaceType {
+                     cmdArgs = &invoke.Args{Command: string("ADD"), ContainerID: containerInfo.ContainerID,
+                             NetNS: containerInfo.ContainerNetNS, IfName: containerInfo.ContainerIfDev,
+                             Path: cniPath}
+
+                     ipamResult, err = ipam.GetIPAMSubnetAddress(cniconfig, cmdArgs)
+                     if err != nil {
+                            return errors.New("secondary network IPAM failed")
+                     }
+
+                     result := &current.Result{CNIVersion: containerInfo.CNIVersion}
+                     result.IPs = ipamResult.IPs
+                     result.Routes = ipamResult.Routes
+	             // Set result.Interface to container interface.
+                     for _, ip := range result.IPs {
+                           ip.Interface = current.Int(1)
+                     }
+                     // If secondary interface name not provided in Pod annotation, assign default name.
+                     if len(netinfo.InterfaceName) == 0 {
+                           netinfo.InterfaceName = netDev
+                     }
+		     // Configure SRIOV as a secondary network interface
+		     if err = pc.configureSriovAsSecondaryInterface(pod, netinfo, containerInfo, result); err != nil {
+			  // Sriov interface creation failed. Free allocated IP address
+                          ipam.DelIPAMSubnetAddress(cniconfig, cmdArgs)
+			  return err
+                     }
+               }
+        }
+	return nil
+}
+
+func (pc *PodController) configureSecondaryNetwork(pod *corev1.Pod, networklist []*SecondaryNetworkObject, podCNIInfo []*cnipodcache.CNIConfigInfo) error {
+
+        for _, netinfo := range networklist {
+               klog.InfoS("Secondary Network Information:", netinfo)
+               if len(netinfo.NetworkName) > 0 {
+                        netDefCRD, err := pc.netAttachDefClient.NetworkAttachmentDefinitions(pod.Namespace).Get(context.TODO(), netinfo.NetworkName, metav1.GetOptions{})
+                        if err != nil {
+                               klog.ErrorS(err, "NetworkAttachmentDefinitions.Get failed", "pod", klog.KObj(pod))
+			       // NetworkAttachmentDefinition not found at this time. return error to re-queue and re-try.
+                               return err
+                        }
+                        cniconfig, err := netdefutils.GetCNIConfig(netDefCRD, "")
+                        if err != nil {
+                               klog.ErrorS("net-attach-def: CNI config spec read error %v", err)
+			       // NetworkAttachmentDefinition Spec.Config parsing failed. return error to re-queue and re-try.
+                               return err
+                        }
+			// secondary network information retrived from API server. Proceed to configure secondary interface now.
+			pc.configureSecondaryInterface(pod, netinfo, podCNIInfo, cniconfig)
```

Comment:
Let us log here if an error is returned, and add a comment saying we wont retry for interface creation errors.